Userland/Libraries/LibGfx/VectorN.h

```diff
@@ -0,0 +1,235 @@
+    }
+
+    [[nodiscard]] constexpr VectorN operator/(const VectorN& other) const
+    {
+        VectorN result {};
+        UNROLL_LOOP
+        for (auto i = 0u; i < N; ++i) {
+            result.m_data[i] = m_data[i] / other.m_data[i];
+        }
+        return result;
+    }
+
+    template<typename U>
+    [[nodiscard]] constexpr VectorN operator*(U f) const
+    {
+        VectorN result {};
+        UNROLL_LOOP
+        for (auto i = 0u; i < N; ++i) {
+            result.m_data[i] = m_data[i] * f;
+        }
+        return result;
+    }
+
+    template<typename U>
+    [[nodiscard]] constexpr VectorN operator/(U f) const
+    {
+        VectorN result {};
+        UNROLL_LOOP
+        for (auto i = 0u; i < N; ++i) {
+            result.m_data[i] = m_data[i] / f;
+        }
+        return result;
+    }
+
+    [[nodiscard]] constexpr T dot(const VectorN& other) const
+    {
+        T result {};
+        UNROLL_LOOP
+        for (auto i = 0u; i < N; ++i) {
+            result += m_data[i] * other.m_data[i];
+        }
+        return result;
+    }
+
+    [[nodiscard]] constexpr VectorN cross(const VectorN& other) const requires(N == 3)
+    {
+        return VectorN(
+            y() * other.z() - z() * other.y(),
+            z() * other.x() - x() * other.z(),
+            x() * other.y() - y() * other.x());
+    }
+
+    [[nodiscard]] constexpr VectorN normalized() const
+    {
+        VectorN copy { *this };
+        copy.normalize();
+        return copy;
+    }
+
+    [[nodiscard]] constexpr VectorN clamped(T m, T x) const
+    {
+        VectorN copy { *this };
+        copy.clamp(m, x);
+        return copy;
+    }
+
+    constexpr void clamp(T min_value, T max_value)
+    {
+        UNROLL_LOOP
+        for (auto i = 0u; i < N; ++i) {
+            m_data[i] = max(min_value, m_data[i]);
+            m_data[i] = min(max_value, m_data[i]);
+        }
+    }
+
+    constexpr void normalize()
+    {
+        T const inv_length = 1 / length();
+        operator*=(inv_length);
+    }
+
+    [[nodiscard]] constexpr T length() const
+    {
+        if constexpr (N == 2) {
+            return AK::hypot(m_data[0] * m_data[0] + m_data[1] * m_data[1]);
+        } else if constexpr (N == 3) {
+            return AK::sqrt(m_data[0] * m_data[0] + m_data[1] * m_data[1] + m_data[2] * m_data[2]);
+        } else {
+            return AK::sqrt(m_data[0] * m_data[0] + m_data[1] * m_data[1] + m_data[2] * m_data[2] + m_data[3] * m_data[3]);
+        }
+    }
+
+    [[nodiscard]] constexpr VectorN<3, T> xyz() const requires(N == 4)
+    {
+        return VectorN<3, T>(x(), y(), z());
+    }
+
+    [[nodiscard]] String to_string() const
+    {
+        if constexpr (N == 2) {
+            return String::formatted("[{},{}]", x(), y());
+        } else if constexpr (N == 3) {
+            return String::formatted("[{},{},{}]", x(), y(), z());
+        } else {
+            return String::formatted("[{},{},{},{}]", x(), y(), z(), w());
+        }
```

Comment:
No braces are required for single line conditionals and loops; see examples in CodingStyle.md. You have quite a number of them in `VectorN` so the code condenses quite a bit (which is arguably better).

Reply:
Done.